src/Poseidon/GenotypeData.hs

```diff
@@ -355,3 +368,59 @@ printSNPCopyProgress logA startTime = do
 
 selectIndices :: [Int] -> (EigenstratSnpEntry, GenoLine) -> (EigenstratSnpEntry, GenoLine)
 selectIndices indices (snpEntry, genoLine) = (snpEntry, V.fromList [genoLine V.! i | i <- indices])
+
+writeVCF :: (MonadSafe m) => LogA -> [JannoRow] -> FilePath -> Consumer (EigenstratSnpEntry, GenoLine) m ()
+writeVCF logA jannoRows vcfFile = do
+    let sampleNames = map (B.pack . jPoseidonID) jannoRows
+        groupNames  = map ((\(GroupName n) -> T.unpack n) . head . getListColumn . jGroupName) jannoRows
+    forM_ jannoRows $ \jannoRow -> do
+        when (jGenotypePloidy jannoRow == Nothing) . logWithEnv logA . logWarning $
+            "Missing GenotypePloidy for individual ++ " ++ jPoseidonID jannoRow ++
+            ". For VCF output I will assume diploid genotypes. " ++
+            "Please set the GenotypePloidy column explitly in the Janno File to Haploid or Diploid."
```

Comment:
better:
> Please set the Genotype_Ploidy column explitly in the .janno file to haploid or diploid.

To get the column and value names right: `Genotype_Ploidy`, `haploid`, `diploid`.

Reply:
Absolutely. Done.